torch/_jit_internal.py

```diff
@@ -159,12 +159,90 @@ def fn(*args, **kwargs):
     }
     return fn
 
+def get_class_attribute_dict(fn, frames_up=2):
+    """
+    Get the frame for class attribute definitions (the top level frame
+    containing methods and class variables defined outside methods)
+    """
+    # default to 2 frames up:
+    #   0 - get_class_attribute_dict
+    #   1 - some decorator
+    #   2 - the class definition
+    frame = inspect.currentframe()
+    i = 0
+    while i < frames_up:
+        frame = frame.f_back
+        i += 1
 
-def ignore(fn):
-    ignored_fns.add(fn)
+    # The locals of the class definition frame contain any class attributes
+    # (such as __constants__)
+    return frame.f_locals
+
+def export(fn):
+    """
+    This decorator indicates that a method is used as an entry point into a
+    ScriptModule. `forward` implicitly is used as an entry point, so it does
+    not need this decorator.
+
+    Methods are added to a ScriptModule as they are called in Python. If a
+    method is never called, it will not be included in the ScriptModule when
+    saving. This decorator explicitly marks that a method should be included
+    even if it is not called from Python.
+    """
+    class_dict = get_class_attribute_dict(fn, frames_up=2)
+    if '__torchscript_export__' not in class_dict:
+        class_dict['__torchscript_export__'] = set()
+    class_dict['__torchscript_export__'].add(fn.__name__)
     return fn
 
 
+def ignore(drop_on_export=False):
```

Comment:
code might be clearer with this declaration:
`ignore(maybe_fn=None, *, drop_on_export=False)`